### ISSUE TITLE
Can't run unit tests involving TokenResponse since version 3.10.10

### ISSUE THREAD

**Issue by ronenfe** [DuendeArchive/IdentityModel#377]
Since this version , it's not possible to instantiate a TokenResponse using
something like that:
```
 tokenClient.RequestClientCredentialsAsync()
                .Returns(c => new TokenResponse(HttpStatusCode.OK, string.Empty, dummyToken));
```
Because the constructor doesn't exist anymore and the properties are readonly.
Any solution?

It means I can't mock RequestClientCredentialsAsync.

**Comment by leastprivilege** [comment]
Maybe - check our unit tests as an example...

**Comment by ronenfe** [comment]
I rephrased the question, I need to mock RequestClientCredentialsAsync in order to unit test my function.
How is it possible in new versions of IdentityModel?
The unit tests in IdentityModel code only tests RequestClientCredentialsAsync itself.

**Comment by leastprivilege** [comment]
Give the factory methods a try

https://github.com/IdentityModel/IdentityModel/blob/072431019a9e9c33be44159ec7f9f7e66c0f8793/src/Client/Messages/ProtocolResponse.cs#L25

**Comment by ronenfe** [comment]
Thanks, I did what you suggested and changed the code to:
```
 private readonly HttpResponseMessage _httpResponse = new HttpResponseMessage(HttpStatusCode.OK);
 tokenClient.RequestClientCredentialsAsync().Returns(c => ProtocolResponse.FromHttpResponseAsync<TokenResponse>(_httpResponse));
```
for the constructor with the 3 parameters and didn't use the second and third parameters of the constructors as I see they are not used by your code anymore.

Also I used:
`var token = ProtocolResponse.FromException<TokenResponse>(new Exception());`
instead of the constructor that had an exception as a parameter. It seems to work fine.
